test/extended/kubevirt/util.go

```diff
@@ -0,0 +1,317 @@
+	podList, err := framework.ClientSet.CoreV1().Pods(framework.Namespace.Name).List(context.Background(), metav1.ListOptions{})
+	Expect(err).To(Succeed())
+	for _, pod := range podList.Items {
+		if strings.Contains(pod.Name, nodeName) {
+			return &pod, nil
+		}
+	}
+
+	return nil, fmt.Errorf("kubevirt node's infra pod not found")
+}
+
+func getSchedulableInfraNode(framework *e2e.Framework, serverInfraNode string) (string, error) {
+
+	// find an infra node to schedule the client on
+	nodes, err := e2enode.GetReadySchedulableNodes(framework.ClientSet)
+	if err != nil {
+		e2e.Logf("Unable to get schedulable nodes due to %v", err)
+		return "", err
+	}
+
+	for _, node := range nodes.Items {
+		if node.Name != serverInfraNode {
+			return node.Name, nil
+		}
+	}
+	return "", fmt.Errorf("infra node not found")
+}
+
+func getNodeInternalAddress(node *v1.Node) (string, error) {
+
+	for _, addr := range node.Status.Addresses {
+
+		if addr.Type == v1.NodeInternalIP {
+			return addr.Address, nil
+		}
+	}
+	return "", fmt.Errorf("no internal node ip found for node %s", node.Name)
+}
+
+func checkKubeVirtGuestClusterHostNetworkNodePortConnectivity(serverFramework, clientFramework *e2e.Framework) error {
+	return checkKubeVirtGuestClusterConnectivity(serverFramework, clientFramework, true, v1.ServiceTypeNodePort)
+}
+
+func checkKubeVirtGuestClusterPodNetworkNodePortConnectivity(serverFramework, clientFramework *e2e.Framework) error {
+	return checkKubeVirtGuestClusterConnectivity(serverFramework, clientFramework, false, v1.ServiceTypeNodePort)
+}
+
+func checkKubeVirtInfraClusterConnectivity(serverFramework, clientFramework *e2e.Framework, oc *exutil.CLI, serviceType v1.ServiceType) error {
+	networking.MakeNamespaceScheduleToAllNodes(serverFramework)
+
+	serverGuestNode, err := getSchedulableNode(serverFramework)
+	Expect(err).To(Succeed())
+
+	serverVMPod, err := getKubeVirtPodFromGuestNode(clientFramework, serverGuestNode)
+	Expect(err).To(Succeed())
+
+	serverInfraNode := serverVMPod.Spec.NodeName
+	serverGuestNodeIP := serverVMPod.Status.PodIP
+
+	infraClientNode, err := getSchedulableInfraNode(clientFramework, serverInfraNode)
+	Expect(err).To(Succeed())
+
+	podName := names.SimpleNameGenerator.GenerateName("service-")
+	defer serverFramework.ClientSet.CoreV1().Pods(serverFramework.Namespace.Name).Delete(context.Background(), podName, metav1.DeleteOptions{})
+	defer serverFramework.ClientSet.CoreV1().Services(serverFramework.Namespace.Name).Delete(context.Background(), podName, metav1.DeleteOptions{})
+
+	serviceAddr := ""
+	if serviceType == v1.ServiceTypeNodePort {
+		serverGuestNodePort := launchWebserverNodePortService(serverFramework.ClientSet, serverFramework.Namespace.Name, podName, serverGuestNode)
+		serviceAddr = net.JoinHostPort(serverGuestNodeIP, strconv.Itoa(int(serverGuestNodePort)))
+	}
+	if serviceType == v1.ServiceTypeLoadBalancer {
+		externalIP := strings.TrimLeft(strings.Split(serverFramework.ClientConfig().Host, ":")[1], "/")
```

Comment:
If you're parsing a URL, use a URL-parsing function. (And if not, at least use `net.SplitHost` rather than `strings.Split`)